### PR TITLE
Test: Replace t.error/fatal with assert/request in [raft_test.go]

### PR DESCRIPTION
**Pull request by MrDXY**
Refactor: Replacing t.Error/t.Fatal with assert/require

This commit tries to eliminates boilerplate code in the testing suite by replacing occurrences
of t.Error and t.Fatal with assertions from the assert and require packages. 

Changes Made:
- Replaced instances of t.Error with assert
- Replaced instances of t.Fatal with require

Affected Files:
- [raft_test.go]

Part of: #146

### CODE REVIEW

File: raft_test.go [hunk 1]
```diff
@@ -2256,15 +1895,9 @@ func TestDisruptiveFollowerPreVote(t *testing.T) {
 	// n1.state == StateLeader
 	// n2.state == StateFollower
 	// n3.state == StateFollower
```
Comment (pav-kv) [inline raft_test.go:1897]: This comment is now unnecessary - the code below is self-explanatory. Remove it?

Same in a few cases below and above.

File: raft_test.go [hunk 2]
```diff
@@ -1793,21 +1536,16 @@ func testCandidateSelfVoteAfterLostElection(t *testing.T, preVote bool) {
...
-		t.Errorf("state = %v, want %v", sm.state, StateFollower)
-	}
+	assert.Equal(t, StateFollower, sm.state)
 
 	// Its self-vote does not make its way to its ProgressTracker.
 	granted, _, _ := sm.trk.TallyVotes()
-	if granted != 0 {
-		t.Errorf("granted = %v, want %v", granted, 0)
-	}
+	assert.Equal(t, 0, granted)
```
Comment (pav-kv) [inline raft_test.go:1547]: `assert.Zero`?

File: raft_test.go [hunk 3]
```diff
@@ -297,34 +267,22 @@ func TestUncommittedEntryLimit(t *testing.T) {
...
 	ms = r.readMessages()
-	if e := 2 * numFollowers; len(ms) != e {
-		t.Fatalf("expected %d messages, got %d", e, len(ms))
-	}
+	require.Equal(t, 2*numFollowers, len(ms))
 	r.reduceUncommittedSize(payloadsSize(propEnts))
-	if n := r.uncommittedSize; n != 0 {
-		t.Fatalf("expected zero uncommitted size, got %d", n)
-	}
+	require.EqualValues(t, 0, r.uncommittedSize)
```
Comment (pav-kv) [inline raft_test.go:285]: `require.Zero`?

Same in a few other places.

File: raft_test.go [hunk 4]
```diff
@@ -268,27 +245,20 @@ func TestUncommittedEntryLimit(t *testing.T) {
...
 
 	// Read messages and reduce the uncommitted size as if we had committed
-	// these entries.
+	// these entries
 	ms := r.readMessages()
-	if e := maxEntries * numFollowers; len(ms) != e {
-		t.Fatalf("expected %d messages, got %d", e, len(ms))
-	}
+	e := maxEntries * numFollowers
+	require.Equal(t, e, len(ms))
```
Comment (pav-kv) [inline raft_test.go:259]: `require.Len`?

Inline `e`?

File: raft_test.go [hunk 5]
```diff
@@ -268,27 +245,20 @@ func TestUncommittedEntryLimit(t *testing.T) {
...
 	}
 
 	// Send one more proposal to r1. It should be rejected.
-	if err := r.Step(propMsg); err != ErrProposalDropped {
-		t.Fatalf("proposal not dropped: %v", err)
-	}
+	require.Equal(t, ErrProposalDropped, r.Step(propMsg))
 
 	// Read messages and reduce the uncommitted size as if we had committed
-	// these entries.
```
Comment (pav-kv) [inline raft_test.go:283]: nit: leave the comment intact?

**Review by pav-kv — APPROVED** [review]
LGTM, with a few nits/suggestions.

**Comment by MrDXY** [comment]
Thank you for your review. I must admit that I may have been a bit hasty in submitting this PR with multiple errors and inconsistencies. And I'll try to double-check everything before the next commit.

File: raft_test.go [hunk 6]
```diff
@@ -489,13 +427,10 @@ func testLeaderCycle(t *testing.T, preVote bool) {
...
-				t.Errorf("preVote=%v: campaigning node %d state = %v, want StateLeader",
-					preVote, sm.id, sm.state)
-			} else if sm.id != campaignerID && sm.state != StateFollower {
-				t.Errorf("preVote=%v: after campaign of node %d, "+
-					"node %d had state = %v, want StateFollower",
-					preVote, campaignerID, sm.id, sm.state)
+			if sm.id == campaignerID {
+				assert.Equal(t, StateLeader, sm.state, "preVote=%v: campaigning node %d", preVote, sm.id)
+			} else {
+				assert.Equal(t, StateFollower, sm.state, "preVote=%v: campaigning node %d, current node %d", preVote, campaignerID, sm.id)
```
Comment (ahrtr) [inline raft_test.go:433]: Nice minor refactor & simplification!

File: raft_test.go [hunk 7]
```diff
@@ -615,47 +536,26 @@ func testVoteFromAnyState(t *testing.T, vt pb.MessageType) {
...
-				t.Errorf("%s,%s: term %d, want %d", vt, st, r.Term, origTerm)
-			}
+			assert.Equal(t, st, r.state, "%s,%s", vt, st)
+			assert.Equal(t, origTerm, r.Term, "%s,%s", vt, st)
 			// if st == StateFollower or StatePreCandidate, r hasn't voted yet.
 			// In StateCandidate or StateLeader, it's voted for itself.
-			if r.Vote != None && r.Vote != 1 {
-				t.Errorf("%s,%s: vote %d, want %d or 1", vt, st, r.Vote, None)
-			}
+			assert.True(t, r.Vote == None || r.Vote == 1, "%s,%s", vt, st)
```
Comment (ahrtr) [inline raft_test.go:558]:
```suggestion
			assert.True(t, r.Vote == None || r.Vote == 1, "%s,%s: vote %d, want %d or 1", vt, st, r.Vote, None)
```

**Review by ahrtr — APPROVED** [review]
lgtm

Thank you!

**Pull request merged**